Tasks/KubernetesManifestV0/src/actions/promote.ts

```diff
@@ -3,26 +3,42 @@ import * as tl from 'azure-pipelines-task-lib/task';
 
 import * as deploymentHelper from '../utils/DeploymentHelper';
 import * as canaryDeploymentHelper from '../utils/CanaryDeploymentHelper';
+import * as SMICanaryDeploymentHelper from '../utils/SMICanaryDeploymentHelper';
 import * as utils from '../utils/utilities';
 import * as TaskInputParameters from '../models/TaskInputParameters';
 
 import { Kubectl } from 'kubernetes-common-v2/kubectl-object-model';
 
 export async function promote(ignoreSslErrors?: boolean) {
-
     const kubectl = new Kubectl(await utils.getKubectl(), TaskInputParameters.namespace, ignoreSslErrors);
 
-    if (canaryDeploymentHelper.isCanaryDeploymentStrategy()) {
-        tl.debug('Deploying input manifests');
-        await deploymentHelper.deploy(kubectl, TaskInputParameters.manifests, 'None');
-        tl.debug('Deployment strategy selected is Canary. Deleting canary and baseline workloads.');
-        try {
-            canaryDeploymentHelper.deleteCanaryDeployment(kubectl, TaskInputParameters.manifests);
-        } catch (ex) {
-            tl.warning('Exception occurred while deleting canary and baseline workloads. Exception: ' + ex);
-        }
-    } else {
+    if (!canaryDeploymentHelper.isCanaryDeploymentStrategy()) {
         tl.debug('Strategy is not canary deployment. Invalid request.');
         throw (tl.loc('InvalidPromotetActionDeploymentStrategy'));
     }
+
+    let includeServices = false;
+    if (canaryDeploymentHelper.isSMICanaryStrategy()) {
+        includeServices = true;
+        // In case of SMI traffic split strategy when deployment is promoted, first we will redirect traffic to
+        // Canary deployment, then update stable deployment and then redirect traffic to stable deployment
+        tl.debug('Redirecting traffic to canary deployment');
+        SMICanaryDeploymentHelper.redirectTrafficToCanaryDeployment(kubectl, TaskInputParameters.manifests);
```

Comment:
Just want to confirm, Redirecting entire traffic to canary during promote, how are we making sure Canary pods can handle that much traffic?

Reply:
In SMI, first you will redirect traffic to other deploymnt(canary) before updating stable. Generally customer will increase the canary pods before updating stable one.